packages/client/hmi-client/src/components/documents/tera-document-asset.vue

```diff
@@ -171,25 +179,27 @@ async function openTextDocument() {
 	const filename: string | undefined = doc.value?.fileNames?.at(0);
 	const res: string | null = await getDocumentFileAsText(props.assetId!, filename!);
 	if (!res) return;
-	code.value = res;
+	docText.value = res;
 }
 
 watch(
 	() => props.assetId,
 	async () => {
 		if (props.assetId) {
 			const document = await getDocumentAsset(props.assetId);
-			if (document) {
-				doc.value = document;
-				openTextDocument();
-				if (viewOptions.value.length > 1) {
-					viewOptions.value.pop();
+			if (!document) {
+				return;
+			}
+			doc.value = document;
+			if (doc.value?.fileNames?.at(0)?.endsWith('.pdf')) {
+				if (view.value === DocumentView.TXT) {
+					view.value = DocumentView.PDF;
+				}
+			} else {
+				await openTextDocument();
+				if (view.value === DocumentView.PDF) {
+					view.value = DocumentView.TXT;
 				}
```

Comment:
I don't quite understand the if-block logic here.

If we open up a file that ends with "pdf", why do we need to check what is the current view-value instead of just directly changing it to PDF?  If we have more view-types like markdown or code, wouldn't this make the logic quite complicated?

Reply:
There are three DocumentView values - PDF, TXT, and EXTRACTIONS. The logic here is to ensure the switcher stays on the same "side" when you switch between documents. I.e. if it was on extractions, it stays on extractions. If it was on PDF/TXT, it stays on PDF/TXT, depending on file type.

Reply:
If the idea is to avoid "resetting" `view.value` then this pattern might be less verbose.

```
if (file_is_pdf) {
  if (view.value !== PDF) view.value = PDF
} else if (file_is_text) {
  if (veiw.value !== TEXT) view.value = TEXT
}
```

Reply:
That doesn't quite work because if `view.value === DocumentView.EXTRACTIONS`, the view will be switched to pdf or text view rather than remaining on extractions. And it's not any less verbose other than not having any braces and personally I prefer using braces even for single line if statements :P

Reply:
Okay - I think there are some convoluted business logic here that I am not aware of. We can take this on later if needed.